pkg/api/types.go

```diff
@@ -1037,6 +1037,18 @@ func (p *Properties) IsUbuntuDistroForAllNodes() bool {
 	return p.MasterProfile.Distro == Ubuntu || p.MasterProfile.Distro == Ubuntu1804
 }
 
+// IsUbuntuForAllNodes returns true if all of the agent pools plus masters are running an Ubuntu-based OS
+func (p *Properties) IsUbuntuForAllNodes() bool {
```

Comment:
All these new "is..." functions are getting confusing IMO... This one is very similar to `IsUbuntuDistroForAllNodes` even though it means something very different. Can we do anything to either make the names more descriptive/distinct or reuse logic instead of having one function for each? eg. `func (p *Properties) IsDistroForAllNodes(distros []Distro) bool {` with a better name.

Reply:
I don't mind doing this refactor in a follow up PR as a proposal and you can tell me if it makes sense.

Reply:
Also `IsUbuntu` does:

```
// IsUbuntu returns true if the master profile distro is any ubuntu distro
func (a *AgentPoolProfile) IsUbuntu() bool {
	return a.IsUbuntu1604() || a.IsUbuntu1804()
}
```
Which is different than what we want here I believe.

Reply:
Alternatively we can get rid of this method and wrap the entire validation script inside an "is my OS ubuntu" if condition. Would you prefer that?

Reply:
Yes definitely.